flytekit/common/mixins/registerable.py

```diff
@@ -73,3 +80,55 @@ def instantiated_in(self):
         :rtype: Optional[Text]
         """
         return self._instantiated_in
+
+    @property
+    def has_valid_name(self):
+        """
+        :rtype: bool
+        """
+        return self._platform_valid_name is not None and self._platform_valid_name != ""
+
+    @property
+    def platform_valid_name(self):
+        """
+        :rtype: Text
+        """
+        return self._platform_valid_name
+
+    def auto_assign_name(self):
+        """
+        This function is a bit of trickster Python code that goes hand in hand with the _InstanceTracker metaclass
+        defined above. Thanks @matthewphsmith for this bit of ingenuity.
+
+        For instance, if a user has code that looks like this:
+
+            from some.other.module import wf
+            my_launch_plan = wf.create_launch_plan()
+
+            @dynamic_task
+            def sample_task(wf_params):
+                yield my_launch_plan()
+
+        This code means that we should have a launch plan with a name ending in "my_launch_plan", since that is the
+        name of the variable that the created launch plan gets assigned to. That is also the name that the launch plan
+        would be registered with.
+
+        However, when the create_launch_plan() function runs, the Python interpreter has no idea where the created
+        object will be assigned to. It has no idea that the output of the create_launch_plan call is to be paired up
+        with a variable named "my_launch_plan". This function basically does this after the fact. Leveraging the
+        _instantiated_in field provided by the _InstanceTracker class above, this code will re-import the
+        module (ie Python file) that the object is in. Since it's already loaded, it's just retrieved from memory.
+        It then scans all objects in the module, and when an object match is found, it knows it's found the right
+        variable name.
+        """
+        _logging.debug("Running name auto assign")
+        m = _importlib.import_module(self.instantiated_in)
+
+        for k in dir(m):
+            if getattr(m, k) == self:
+                _logging.debug("Auto-assigning name to {}".format(k))
+                self._platform_valid_name = "{}.{}".format(self.instantiated_in, k)
```

Comment:
there is a util function somewhere that does this (and it is important for our legacy systems at Lyft because they use a different naming convention). Search for `fqdn`, I think.

Reply:
I was following this: https://github.com/lyft/flytekit/blob/master/flytekit/common/tasks/sdk_runnable.py#L233

Reply:
Nm, I think I got it.  You mean that when a launch plan is registered right now, the name that is assigned to it comes from register.py, which uses the fqdn function.
https://github.com/lyft/flytekit/pull/92/commits/76a812cd3ba43353f2ed040d4b5b4ae912554f4b